tests/tests_purge_reset.yml

```diff
@@ -1,11 +1,86 @@
 # Test to ensure the logging role successfully runs with
 # logging_purge_confs and logging_reset_confs set to true.
 #
-- name: Ensure that the role runs with default parameters
+- name: Test purge and reset functionality
   hosts: all
-
+  vars:
+    __test_default_files_conf: /etc/rsyslog.d/30-output-files-default_files.conf
   tasks:
-    - name: default run with purge and reset
+    - name: ensure rsyslog is installed
+      package:
+        name: rsyslog
+        state: present
+
+    - name: create some unowned config files
+      copy:
+        dest: "{{ item }}"
+        content: "# this is file {{ item }}"
+        mode: "0444"
+      loop:
+        - "/etc/rsyslog.d/unowned1.conf"
+        - "/etc/rsyslog.d/unowned2.conf"
+
+    - name: modify /etc/rsyslog.conf
+      lineinfile:
+        path: /etc/rsyslog.conf
+        insertafter: EOF
+        state: present
+        line: "# this is a modification"
+
+    - name: configure with basics + purge + reset
+      vars:
+        logging_purge_confs: true
+        logging_reset_confs: true
+        logging_inputs:
+          - name: basic_input
+            type: basics
+      include_role:
+        name: linux-system-roles.logging
+
+    # notify restart rsyslogd is executed at the end of this test task.
+    # thus we have to force to invoke handlers
+    - name: "Force all notified handlers to run at this point,
+      not waiting for normal sync points"
+      meta: flush_handlers
+
+    - name: Ensure config file size and counts
+      vars:
+        __conf_count: 5
+        __conf_size: less
+        __conf_files:
+          - "{{ __test_default_files_conf }}"
+        __check_systemctl_status: true
+      include_tasks: tasks/check_daemon_config_files.yml
+
+    - name: Ensure previous test config files are removed
+      stat:
+        path: "{{ item }}"
+      register: __result
+      failed_when: __result.stat.exists
+      loop:
+        - "/etc/rsyslog.d/unowned1.conf"
+        - "/etc/rsyslog.d/unowned2.conf"
+
+    - name: check1 rsyslog config files
+      command: ls /etc/rsyslog.d
+      changed_when: false
+
+    # HACK HACK HACK
+    # rsyslog uses set_fact which creates global variables which
+    # carry over to subsequent runs of the role in the same process
+    # so we have to wipe them out, otherwise, the configuration set
+    # above will be reapplied when we try to purge/reset with no
+    # config
+    - name: HACK - reset global variables
+      set_fact:
+        __rsyslog_output_files: []
+        rsyslog_outputs: []
+        rsyslog_inputs: []
+        rsyslog_custom_config_files: []
+        __rsyslog_common_rules: []
+    # HACK HACK HACK
+
+    - name: run again with purge + reset to erase all config
       vars:
         logging_purge_confs: true
         logging_reset_confs: true
```

Comment:
Actually, eliminating the "HACK - reset global variables" task and add the infamous `logging_enabled: false` to this task does the trick. ;)
```
     - name: run again with purge + reset to erase all config
       vars:
+        logging_enabled: false
         logging_purge_confs: true
         logging_reset_confs: true
       include_role:
```

Reply:
hmm - ok - note that this is only necessary because we ran the role with actual settings previously in the same process - if a user wants to use reset to revert back to the system defaults, they will have to do it in a separate run of ansible-playbook, or those global variables will still be set (or use logging_enabled: false)

Reply:
I have another branch which gets rid of the global variables - but I think that change is too invasive for the purge/reset issue

Reply:
> hmm - ok - note that this is only necessary because we ran the role with actual settings previously in the same process - if a user wants to use reset to revert back to the system defaults, they will have to do it in a separate run of ansible-playbook, or those global variables will still be set (or use logging_enabled: false)

I agree. The logging role tests try to revert the config files as much as possible in "END TEST CASE #; Clean up the deployed config" to allow to run multiple tests repeatedly in one VM, in which the same global variables are given with `state: absent` to delete the previously generated files.  (Right. `/etc/rsyslog.conf` is not reverted. So, the configuration is almost empty and useless as `rsyslog`.) The cleanup code could be replaced with:
```
       vars:
+        logging_enabled: false
         logging_purge_confs: true
         logging_reset_confs: true
```
Having the simple(?) reset method would be useful for our CI which repeats a series of tests. (Not sure how it is in the real use cases, though...)